nemo/deploy/tritondeployables/megatrongpt_deployable.py

```diff
@@ -0,0 +1,122 @@
+import logging
+from pathlib import Path
+
+import numpy as np
+import wrapt
+from pytorch_lightning.trainer.trainer import Trainer
+
+from nemo.collections.nlp.models.language_modeling.megatron_gpt_model import MegatronGPTModel
+from nemo.collections.nlp.modules.common.text_generation_utils import (
+    get_computeprob_response,
+    get_default_length_params,
+    get_default_sampling_params,
+)
+from nemo.collections.nlp.modules.common.transformer.text_generation import LengthParam, SamplingParam
+from nemo.collections.nlp.parts.nlp_overrides import NLPDDPStrategy, NLPSaveRestoreConnector
+from nemo.deploy import ITritonDeployable
+from nemo.deploy.utils import cast_output, str_ndarray2list
+
+
+@wrapt.decorator
+def noop_decorator(func):
+    def wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+use_pytriton = True
+batch = noop_decorator
+try:
+    from pytriton.decorators import batch
+    from pytriton.model_config import Tensor
+except Exception:
+    use_pytriton = False
+
+LOGGER = logging.getLogger("NeMo")
+
+
+class MegatronGPTDeployable(ITritonDeployable):
```

Comment:
Let's change this name to PyTritonDeployableLLM and move this file under nemo/deploy/nlp

Reply:
Renamed to `MegatronLLMDeployable` for now.  As discussed offline, whatever name we decide on for this class should match the Model class that we create inside of it.  So if the final name is `MegatronLLMDeployable`, the internal self.model should be of type `MegatronLLMModel`.